TokenRing/src/main/java/RingProcessor.java

```diff
@@ -0,0 +1,122 @@
+ */
+public class RingProcessor {
+    private final int nodesAmount;
+    private final int dataAmount;
+
+    private final File logs;
+
+    private List<Node> nodeList;
+
+    private final Logger logger;
+    FileHandler fileHandler;
+    ArrayList<Thread> threads;
+
+    /**
+     * A record of the transit time of each data package.
+     * Used in {@link RingProcessor # averageTime ()} to calculate average time
+     * of reaching the coordinator by the data.
+     */
+
+    List<Long> timeList;
+
+    RingProcessor(int nodesAmount, int dataAmount, File logs) throws IOException, InterruptedException {
+        this.nodesAmount = nodesAmount;
+
+        this.dataAmount = dataAmount;
+
+        this.logs = logs;
+        nodeList = new ArrayList<>();
+        logger = Logger.getLogger("ringLogger");
+        fileHandler  = new FileHandler(logs.getName(), true);
+        logger.addHandler(fileHandler);
+        logger.info("Number of Nodes: " + nodesAmount);
+        logger.info("Data on each node: " + dataAmount);
+        init();
+    }
+
+    // Computation of the average traversing time.
+    private long averageTime() {
+        return 0;
+    }
+
+    private void init() throws IOException, InterruptedException {
+        Random rand = new Random();
+        int coordinator = rand.nextInt(nodesAmount) + 1;
+        logger.info("Coordinator Node id: " + coordinator);
+        threads = new ArrayList<>();
+        //initialize ring
+        for (int i = 1; i <= nodesAmount; ++i) {
+            RingProcessor temp = null;
+            if (i == coordinator) {
+                temp = this;
+            }
+            Node newNode = new Node(i, coordinator, dataAmount, temp, fileHandler, logger);
+            nodeList.add(newNode);
+            threads.add(new Thread(newNode, String.valueOf(i)));
+            int receiveNode = ThreadLocalRandom.current().nextInt(nodesAmount) + 1;
+            if (receiveNode == i) {
+                if (receiveNode == nodesAmount) {
+                    receiveNode -= 1;
+                } else {
+                    receiveNode += 1;
+                }
+            }
+
+            DataPackage dataPackage = new DataPackage(receiveNode, generateRandomString(i));
+            newNode.setData(dataPackage);
+            System.out.println(i + " " + dataPackage + " " + receiveNode);
+        }
+
+        for (int i = 0; i < nodesAmount; ++i) {
+            int j = min(i + 1, nodesAmount) == nodesAmount ? 0 : i + 1;
+            nodeList.get(i).addNodeAndCoord(nodeList.get(j), nodeList.get(coordinator - 1));
+        }
+    }
+
+    public void startProcessing() throws InterruptedException {
+        for (int i = 0; i < nodesAmount; ++i) {
+            threads.get(i).start();
+        }
+    }
+
+    public void interruptAllNodes() {
+        System.out.println("Shutting threads down");
+        for (Thread t : threads) {
+            t.interrupt();
+        }
+
+        System.exit(0);

```

Comment:
thst's not good. This method terminates the whole JVM, not just threads. You shut down all nodes threads, and the main thread finishes, as there are no other operations